src/object/notification.c

```diff
@@ -259,6 +278,16 @@ void completeSignal(notification_t *ntfnPtr, tcb_t *tcb)
         notification_ptr_set_state(ntfnPtr, NtfnState_Idle);
 #ifdef CONFIG_KERNEL_MCS
         maybeDonateSchedContext(tcb, ntfnPtr);
+#endif
+#ifdef CONFIG_KERNEL_MCS
+        if (tcb->tcbSchedContext != NULL && sc_active(tcb->tcbSchedContext)) {
```

Comment:
Does this block also need an sc_sporadic() block?

Reply:
I think you're right